web/src/main/java/org/openmrs/web/filter/util/CustomResourceLoader.java

```diff
@@ -92,7 +92,10 @@ private Locale parseLocaleFrom(String filename, String basename) {
 		Locale result = null;
 		
 		if (filename.startsWith(basename)) {
-			filename = filename.substring(basename.length());
+			
+			String tempfilename = filename;
+			tempfilename = tempfilename.substring(basename.length());
+			filename = tempfilename;
 		}
 		
 		String localespec = filename.substring(0, filename.indexOf('.'));
```

Comment:
Closely look at the process flow. Once we introduce the `tempfilename`, the line here also need to be changed,  for the method work correctly.

    String localespec = tempfilename.substring(0, filename.indexOf('.'));